core/services/im_here/MavlinkNMEA.py

```diff
@@ -0,0 +1,86 @@
+from enum import IntEnum
+from typing import Optional
+
+import pynmea2
+from pydantic import BaseModel
+
+from exceptions import UnsupportedSentenceType
+
+
+class GPS_FIX_TYPE(IntEnum):
+    GPS_FIX_TYPE_NO_GPS = 0
+    GPS_FIX_TYPE_NO_FIX = 1
+    GPS_FIX_TYPE_2D_FIX = 2
+    GPS_FIX_TYPE_3D_FIX = 3
+    GPS_FIX_TYPE_DGPS = 4
+    GPS_FIX_TYPE_RTK_FLOAT = 5
+    GPS_FIX_TYPE_RTK_FIXED = 6
+    GPS_FIX_TYPE_STATIC = 7
+    GPS_FIX_TYPE_PPP = 8
+
+
+class GPS_INPUT_IGNORE_FLAG(IntEnum):
+    GPS_INPUT_IGNORE_FLAG_ALT = 1
+    GPS_INPUT_IGNORE_FLAG_HDOP = 2
+    GPS_INPUT_IGNORE_FLAG_VDOP = 4
+    GPS_INPUT_IGNORE_FLAG_VEL_HORIZ = 8
+    GPS_INPUT_IGNORE_FLAG_VEL_VERT = 16
+    GPS_INPUT_IGNORE_FLAG_SPEED_ACCURACY = 32
+    GPS_INPUT_IGNORE_FLAG_HORIZONTAL_ACCURACY = 64
+    GPS_INPUT_IGNORE_FLAG_VERTICAL_ACCURACY = 128
+
+
+class Mavlink2RestBitEnum(BaseModel):
+    bits: int
+
+
+nmea_ignore_flags = (
+    GPS_INPUT_IGNORE_FLAG.GPS_INPUT_IGNORE_FLAG_SPEED_ACCURACY
+    + GPS_INPUT_IGNORE_FLAG.GPS_INPUT_IGNORE_FLAG_VEL_VERT
+    + GPS_INPUT_IGNORE_FLAG.GPS_INPUT_IGNORE_FLAG_VEL_HORIZ
+)
+
+
+class MavlinkGeoPackage(BaseModel):
+    type: str = "GPS_INPUT"
+    time_usec: Optional[int] = 0
+    gps_id: Optional[int] = 0
+    ignore_flags: Optional[Mavlink2RestBitEnum] = Mavlink2RestBitEnum(bits=(nmea_ignore_flags))
+    time_week_ms: Optional[int] = 0
+    time_week: Optional[int] = 0
+    fix_type: Optional[GPS_FIX_TYPE] = GPS_FIX_TYPE.GPS_FIX_TYPE_3D_FIX
+    lat: int
+    lon: int
+    alt: Optional[float] = 0
+    hdop: Optional[float] = 0
+    vdop: Optional[float] = 0
+    vn: Optional[float] = 0
+    ve: Optional[float] = 0
+    vd: Optional[float] = 0
+    speed_accuracy: Optional[float] = 0
+    horiz_accuracy: Optional[float] = 0
+    vert_accuracy: Optional[float] = 0
+    satellites_visible: Optional[int] = 0
+    yaw: Optional[int] = 0
+
+
+def parse_mavlink_from_sentence(msg: pynmea2.NMEASentence) -> MavlinkGeoPackage:
+    data = {}
+
+    supported_sentence_types = ["GGA", "RMC", "GLL", "GNS"]
+    if msg.sentence_type not in supported_sentence_types:
+        raise UnsupportedSentenceType(f"Supported types are {supported_sentence_types}. Received {msg.sentence_type}")
+
+    data["lat"] = msg.latitude * 1e7
+    data["lon"] = msg.longitude * 1e7
```

Comment:
Can you add a comment about the 1e7 value and why it's necessary ?

Reply:
Sure!